### ISSUE TITLE
Determine minimum compatible version of Kubernetes

### ISSUE THREAD

**Issue by krancour** [brigadecore/brigade#1441]
We've used 1.18 - 1.20 throughout Brigade 2 development, but we don't rightfully know what the minimum compatible version of k8s is.

This issue only calls for some research and experimentation to make that determination.

@willie-yao this might be something you can play with as time permits.

cc: @carolynvs-- thanks for raising this issue.

**Comment by willie-yao** [comment]
![image](https://user-images.githubusercontent.com/47200969/121272815-bc436c80-c87b-11eb-8d61-b4104546e968.png)
k8s 1.17.0 keeps crashing 20 minutes in

**Comment by willie-yao** [comment]
![image](https://user-images.githubusercontent.com/47200969/121273220-95396a80-c87c-11eb-91e8-cad06fc0e3f5.png)
1.18.0 works fine

**Comment by willie-yao** [comment]
![image](https://user-images.githubusercontent.com/47200969/121273584-6a9be180-c87d-11eb-9a68-edab4cc1e244.png)
1.17.5 works fine

**Comment by willie-yao** [comment]
![image](https://user-images.githubusercontent.com/47200969/121274549-93bd7180-c87f-11eb-8587-41ab1f766bae.png)
1.17.2 results in infinite crashing loop, making 1.17.5 the last kind node version on dockerhub to work with brigade

**Comment by krancour** [comment]
@willie-yao can you please get logs from whatever's crashing on 1.17.0 and 1.17.2? It would be good if we could pinpoint why we're incompatible. It's also possible the logs reveal something that's easily fixed to ensure broader compatibility.

**Comment by krancour** [comment]
After more research, coredns wasn't coming up clean when using a newer kind CLI and older kind image.

For such scenarios, this hack moved us past this issue, presenting us with a healthy single-node k8s cluster on which to test Brigade 2:

https://stackoverflow.com/questions/53075796/coredns-pods-have-crashloopbackoff-or-error-state/53414041#53414041

We've already found Brigade 2 works fine on a healthy k8s 1.17.2 and @willie-yao will resume the hunt for an older version of k8s that we are legitimately not compatible with.

**Comment by willie-yao** [comment]
As an update to the previous comment, using this hack actually caused a dns issue when running a job with brigade. I resumed testing with minikube instead of kind, which has better support for older k8s versions.

The last compatible k8s version is 1.16.0, as using anything below 1.16.0 causes this issue when installing the helm chart for Brigade2:
`Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(Deployment.spec.template.spec.containers[0]): unknown field "startupProbe" in io.k8s.api.core.v1.Container`

This is because startupProbes were added in k8s version 1.16, thus making 1.16 the minimum compatible version of k8s for Brigade2. I'll be closing the issue now as the question is resolved.
